src/twisted/application/runner/test/test_runner.py

```diff
@@ -273,56 +293,56 @@ def test_startReactorWithReactor(self):
 
         self.assertTrue(reactor.hasRun)
 
-    def test_startReactorWhenRunning(self):
+    def test_startReactorWhenRunning(self) -> None:
         """
         L{Runner.startReactor} ensures that C{whenRunning} is called with
         C{whenRunningArguments} when the reactor is running.
         """
         self._testHook("whenRunning", "startReactor")
 
-    def test_whenRunningWithArguments(self):
+    def test_whenRunningWithArguments(self) -> None:
         """
         L{Runner.whenRunning} calls C{whenRunning} with
         C{whenRunningArguments}.
         """
         self._testHook("whenRunning")
 
-    def test_reactorExitedWithArguments(self):
+    def test_reactorExitedWithArguments(self) -> None:
         """
         L{Runner.whenRunning} calls C{reactorExited} with
         C{reactorExitedArguments}.
         """
         self._testHook("reactorExited")
 
-    def _testHook(self, methodName, callerName=None):
+    def _testHook(self, methodName: str, callerName: Optional[str] = None) -> None:
         """
         Verify that the named hook is run with the expected arguments as
         specified by the arguments used to create the L{Runner}, when the
         specified caller is invoked.
 
         @param methodName: The name of the hook to verify.
-        @type methodName: L{str}
 
         @param callerName: The name of the method that is expected to cause the
             hook to be called.
             If C{None}, use the L{Runner} method with the same name as the
             hook.
-        @type callerName: L{str}
         """
         if callerName is None:
             callerName = methodName
 
         arguments = dict(a=object(), b=object(), c=object())
         argumentsSeen = []
 
-        def hook(**arguments):
+        def hook(**arguments: Any) -> None:
```

Comment:
If you're not making any assumptions at all about the type, you could use `object` instead of `Any`.

Reply:
Hrm.  I tend to use `Any` in those cases/. I feel like `object` is being specific when I mean to be non-specific, no?

Reply:
The way I use them, `object` means "any object and I don't care what operations it implements", while `Any` means "I know something about this object that is too complex to express in Python's type system".

Reply:
Ah.  OK one reason I tend to use `Any` for `**kwargs` in particular is that I often pass those on to some other callable in that form, and if you use `object`, `mypy` might complain that said callable wants something else there.  I prefer to be consistent about that than to start with `object` expect if I actually use it then switch to `Any`.

Looking at this specific example, this really is a "whatever you gave me is fine" scenario; I'm just storing your crap in a list for you to look at later.  I have no opinion about that crap.

But then, the only expected operation on these things for the purposes of the test is… equality, which `object` has.  So… ok.

This resining is a bit subtle, so we should think about how to write it down.

Reply:
It remains unclear to me what the best choice is [here](https://github.com/twisted/twisted/pull/1410/files#diff-6e519d82a6efe6b3cc078b899bbc320fL423), for example.  Still `object`, though I'm YOLO-emulating another API for testing?